src/main/java/manufacturing/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,215 @@
+package manufacturing.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import manufacturing.dao.CarDao;
+import manufacturing.exception.DataProcessingException;
+import manufacturing.lib.Dao;
+import manufacturing.model.Car;
+import manufacturing.model.Driver;
+import manufacturing.model.Manufacturer;
+import manufacturing.util.ConnectionUtil;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars "
+                + "(car_model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject("GENERATED_KEY", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create car "
+                    + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> getById(Long carId) {
+        String query = "SELECT c.car_id, c.car_model, c.manufacturer_id, "
+                + "m.manufacturer_name, m.manufacturer_country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m "
+                + "ON m.manufacturer_id = c.manufacturer_id "
+                + "WHERE car_id = ? AND c.deleted = FALSE";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, carId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                car = createCar(resultSet, connection);
+                car.setDrivers(getAllDriversByCarId(carId, connection));
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get Car by id "
+                    + carId, e);
+        }
```

Comment:
```suggestion
        Car car = null;
        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
            preparedStatement.setLong(1, carId);
            ResultSet resultSet = preparedStatement.executeQuery();
            if (resultSet.next()) {
                car = createCar(resultSet);
            }
        } catch (SQLException e) {
            throw new DataProcessingException("Can't get Car by id "
                    + carId, e);
        }
        if (car != null) {
            car.setDrivers(getAllDriversByCarId(carId));
        }
        return Optional.ofNullable(car);
```

although you passed the connection, your statement is not closed yet, and you opened another statement in it
let's close the current connection and statement and then open new connection in getAllDriversByCarId() method

Reply:
agree. done